File: JAVA/2149. Rearrange Array Elements by Sign.java
class Solution {
    public int[] rearrangeArray(int[] n) {
        int[] a=new int[n.length/2];
        int[] b=new int[n.length/2];
        int i=0,j=0;
        for(int l:n){
            if(l<0)
            b[j++]=l;
            else
            a[i++]=l;
        }
        j=0;
        for(i=0;i<a.length;i++){
            n[j++]=a[i];
            n[j++]=b[i];
        }
        return n;
    }
}

File: C++/2149. Rearrange Array Elements by Sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& n) {
        vector<int>a,b;
        for(int i:n){
            if(i<0)
            b.push_back(i);
            else
            a.push_back(i);
        }
        n.clear();
        for(int i=0;i<a.size();i++){
            n.push_back(a[i]);
            n.push_back(b[i]);
        }
        return n;
    }
};